.github/workflows/build.yml

```diff
@@ -137,7 +140,7 @@ jobs:
           path: "dist/bin/saw"
 
       - shell: bash
-        if: "steps.test.outcome == 'failure'"
+        if: "steps.test.outcome == 'failure' || steps.cabal-test.outcome == 'failure'"
         name: Warn if tests failed
         run: echo "::error ::Test Suite Failed. Pipeline allowed to pass until tests are reliable."
```

Comment:
One way to handle a situation like this is regression testing (indicate failure when tests that were previously passing are now failing), maybe it would be worthwhile to implement something like this that can be used generally for cabal tests?